deploy/templates/service-template.yml

```diff
@@ -0,0 +1,330 @@
+    spec:
+      selector:
+        matchLabels:
+          app: ${DB_SERVICE_NAME}
+      replicas: 1
+      template:
+        metadata:
+          labels:
+            app: ${DB_SERVICE_NAME}
+        spec:
+          containers:
+            - name: postgres
+              image: ${POSTGRES_IMAGE}
+              imagePullPolicy: IfNotPresent
+              ports:
+                - containerPort: 5432
+                  protocol: TCP
+              env:
+                - name: POSTGRESQL_DATABASE
+                  valueFrom:
+                    secretKeyRef:
+                      name: ${DB_SECRET_NAME}
+                      key: db.name
+                - name: POSTGRESQL_USER
+                  valueFrom:
+                    secretKeyRef:
+                      name: ${DB_SECRET_NAME}
+                      key: db.user
+                - name: POSTGRESQL_PASSWORD
+                  valueFrom:
+                    secretKeyRef:
+                      name: ${DB_SECRET_NAME}
+                      key: db.password
+              volumeMounts:
+                - mountPath: /var/lib/pgsql/data
+                  name: ${DB_SERVICE_NAME}-data
+              resources:
+                limits:
+                  memory: 500Mi
+                requests:
+                  cpu: 100m
+                  memory: 400Mi
+              livenessProbe:
+                tcpSocket:
+                  port: 5432
+                initialDelaySeconds: 120
+                timeoutSeconds: 10
+              readinessProbe:
+                exec:
+                  command:
+                  - /bin/sh
+                  - -c
+                  - exec /usr/bin/pg_isready -U $POSTGRESQL_USER -d $POSTGRESQL_DATABASE -h localhost -p 5432
+                initialDelaySeconds: 5
+                timeoutSeconds: 1
+              securityContext:
+                capabilities: {}
+                privileged: false
+              terminationMessagePath: /dev/termination-log
+              capabilities: {}
+          dnsPolicy: ClusterFirst
+          restartPolicy: Always
+          volumes:
+            - name: ${DB_SERVICE_NAME}-data
+              persistentVolumeClaim:
+                claimName: ${DB_SERVICE_NAME}-data
+  - kind: Secret
+    apiVersion: v1
+    metadata:
+      name: ${DB_SECRET_NAME}
+      labels:
+        app: postgres
+    type: Opaque
+    stringData:
+      config.yaml: |
+        database:
+          hostname: ${DB_SERVICE_NAME}
+          name: ${DB_NAME}
+          password: ${DB_PASSWORD}
+          port: 5432
+          type: pgsql
+          user: ${DB_USER}
+        service:
+          address: :3443
+          agentEndpointAddress: :7443
+          baseAgentEndpointUrl: https://localhost:7443
+          baseUrl: https://localhost:3443
+          logLevel: debug
+      db.host: ${DB_SERVICE_NAME}
+      db.name: ${DB_NAME}
+      db.port: "5432"
+      db.user: ${DB_USER}
+      db.password: ${DB_PASSWORD}
```

Comment:
I would create two secret for this. I've just simplified it in k8s deployment scripts, but on app-sre it should be seprate.